command/agent/plugins.go

```diff
@@ -10,6 +10,10 @@ import (
 
 // setupPlugins is used to setup the plugin loaders.
 func (a *Agent) setupPlugins() error {
+	if a.GetConfig().Client == nil || !a.GetConfig().Client.Enabled {
+		a.logger.Debug("skipping plugin setup because node is not a client")
+		return nil
+	}
```

Comment:
Instead of an early return what if we just called this method from within `setupClient`? I just find the conditions a bit confusing because they should always match but may not depending on _when_ you check them.